core/domain/rte_component_registry_test.py

```diff
@@ -174,3 +175,25 @@ def test_default_rte_components_are_valid(self):
 
             self._validate_customization_arg_specs(
                 component._customization_arg_specs)  # pylint: disable=protected-access
+
+    def test_html_contains_all_imports(self):
+        """Test that the rich_text_components.html contains script-imports for
+        all directives of all RTE components."""
+
+        js_files_paths = []
+        for component_id in feconf.ALLOWED_RTE_EXTENSIONS:
+            component_dir = os.path.join(
+                feconf.RTE_EXTENSIONS_DIR, component_id)
+            directives_dir = os.path.join(component_dir, 'directives')
+            names = os.listdir(directives_dir)
+            js_files_paths.extend(
+                os.path.join(directives_dir, name) for name in names
+                if name.endswith('.js'))
+
+        rtc_html_file = os.path.join(
+            feconf.FRONTEND_TEMPLATES_DIR, 'pages', 'rich_text_components.html')
+        with open(rtc_html_file, 'r') as f:
+            rtc_html_file_contents = f.read()
+
+        for js_file_path in js_files_paths:
+            self.assertIn(js_file_path, rtc_html_file_contents)
```

Comment:
Can you check for the whole tag: `<script src="...">`?

Also maybe check exhaustively, e.g. number of lines = 6, each line is one of these, there are no extra lines/imports? I think we should try and fully fix the content of this file. In fact a good way to handle this would be to programmatically generate the file's contents here (just order the components alphabetically so that it's deterministic) and compare what's generated to the file's actual contents.

Reply:
Done.